sdk/python/kfp/utils.py

```diff
@@ -0,0 +1,61 @@
+def use_secret(secret_name:str=None, volume_name:str=None, secret_volume_mount_path:str=None, env_variable:str=None, secret_file_path_in_volume:str=None):
+    """    
+       An operator that configures the container to use a secret.
+       
+       This assumes that the secret is created and availabel in the k8s cluster.
+    
+    Keyword Arguments:
+        secret_name {String} -- [Required] The k8s secret name (default: {None})
+        volume_name {String} -- The pod volume name (default: {None})
+        secret_volume_mount_path {String} -- [Required] The path to the secret that is mounted  (default: {None})
+        env_variable {String} -- Env variable pointing to the mounted secret file. Requires both the env_variable and secret_file_path_in_volume to be defined. 
+                                 The value is the path to the secret (default: {None})
+        secret_file_path_in_volume {String} -- The path to the secret in the volume. This will be the value of env_variable. 
+                                 Both env_variable and secret_file_path_in_volume needs to be set if any env variable should be created (default: {None})
+    
+    Raises:
+        ValueError: If not the necessary variables (secret_name, volume_name", secret_volume_mount_path) are supplied.
+                    Or only one of  env_variable and secret_file_path_in_volume are supplied
+    
+    Returns:
+        [ContainerOperator] -- Returns the container operator after it has been modified. 
+    """
+
+    if secret_name:
+        volume_name = volume_name or secret_name + '_volume'
+
+    params = [secret_name, volume_name, secret_volume_mount_path]
+    param_names = ["secret_name", "volume_name", "secret_volume_mount_path"]
```

Comment:
If we do not default the param to None, can we save the effort of doing this check?

Reply:
I will updated this :)